core/trino-main/src/main/java/io/trino/sql/planner/AnonymizePlanRewriter.java

```diff
@@ -0,0 +1,785 @@
+                node.getDistributionType(),
+                node.getDynamicFilterId());
+    }
+
+    @Override
+    public PlanNode visitSpatialJoin(SpatialJoinNode node, RewriteContext<Void> context)
+    {
+        return new SpatialJoinNode(
+                node.getId(),
+                node.getType(),
+                context.rewrite(node.getLeft()),
+                context.rewrite(node.getRight()),
+                anonymize(node.getOutputSymbols()),
+                anonymize(node.getFilter()),
+                node.getLeftPartitionSymbol().map(this::anonymize),
+                node.getRightPartitionSymbol().map(this::anonymize),
+                node.getKdbTree());
+    }
+
+    @Override
+    public PlanNode visitIndexJoin(IndexJoinNode node, RewriteContext<Void> context)
+    {
+        List<IndexJoinNode.EquiJoinClause> anonymizedCriteria = node.getCriteria().stream()
+                .map(joinClause -> new IndexJoinNode.EquiJoinClause(anonymize(joinClause.getProbe()), anonymize(joinClause.getIndex())))
+                .collect(toImmutableList());
+
+        return new IndexJoinNode(
+                node.getId(),
+                node.getType(),
+                context.rewrite(node.getProbeSource()),
+                context.rewrite(node.getIndexSource()),
+                anonymizedCriteria,
+                node.getProbeHashSymbol().map(this::anonymize),
+                node.getIndexHashSymbol().map(this::anonymize));
+    }
+
+    @Override
+    public PlanNode visitSort(SortNode node, RewriteContext<Void> context)
+    {
+        return new SortNode(
+                node.getId(),
+                context.rewrite(node.getSource()),
+                anonymize(node.getOrderingScheme()),
+                node.isPartial());
+    }
+
+    @Override
+    public PlanNode visitWindow(WindowNode node, RewriteContext<Void> context)
+    {
+        return new WindowNode(
+                node.getId(),
+                context.rewrite(node.getSource()),
+                new Specification(anonymize(node.getPartitionBy()), node.getOrderingScheme().map(this::anonymize)),
+                anonymize(node.getWindowFunctions()),
+                node.getHashSymbol().map(this::anonymize),
+                node.getPrePartitionedInputs(),
+                node.getPreSortedOrderPrefix());
+    }
+
+    @Override
+    public PlanNode visitRefreshMaterializedView(RefreshMaterializedViewNode node, RewriteContext<Void> context)
+    {
+        return new RefreshMaterializedViewNode(
+                node.getId(),
+                new QualifiedObjectName(
+                        anonymize(node.getViewName().getCatalogName()),
+                        anonymize(node.getViewName().getSchemaName()),
+                        anonymize(node.getViewName().getObjectName())));
+    }
+
+    @Override
+    public PlanNode visitTableWriter(TableWriterNode node, RewriteContext<Void> context)
```

Comment:
It looks like there is information needing anonymisation in TableWriterNode as well